src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -2030,17 +2055,21 @@ GenTree* Compiler::impBaseIntrinsic(NamedIntrinsic        intrinsic,
 
             if ((simdSize != 32) || compExactlyDependsOn(InstructionSet_AVX2))
             {
+                genTreeOps op = varTypeIsUnsigned(simdBaseType) ? GT_RSZ : GT_RSH;
```

Comment:
If the base type is unsigned, is `op_RightShift` basically going to be the same thing as `op_UnsignedRightShift`?

Reply:
Correct, that's how `op_RightShift` is expected to behave for unsigned types